tcms/testcases/tests/test_views.py

```diff
@@ -338,6 +338,15 @@ def test_show_clone_page_with_selected_cases(self):
             self.assertContains(response,
                                 "TC-%d: %s" % (case.pk, case.summary))
 
+    def test_user_without_permission_should_not_be_able_to_clone_a_case(self):
+        remove_perm_from_user(self.tester, 'testcases.add_testcase')
+        url = reverse('testcases-clone') + quote('?case=' + str(self.case_1.pk))
+
+        self.assertRedirects(
+            self.client.get(self.clone_url,
```

Comment:
assign this to a `response` variable and assert on the variable to minimize nesting and improve readability.